### PR TITLE
frontend: add simple part of vue/vue3-recommended rules

### PR DESCRIPTION
**Pull request by BacLuc**
Turn off the following rules for now:
"vue/no-deprecated-dollar-scopedslots-api": "off", 
"vue/no-deprecated-slot-attribute": "off",
"vue/no-deprecated-slot-scope-attribute": "off",
because i could not get the new syntax to work.

That we are ready for vue3 as soon as the calendar works. Issue: #5121

### CODE REVIEW

**Review by pmattmann — CHANGES_REQUESTED** [review]
I'm unsure if it has to do with this PR, but I can't create new activities on the Picasso (by dragging).

**Comment by BacLuc** [comment]
Seems that the lint rule rewrites the event identifier where you register the event, but not the place where you emit the event.
Fixed with https://github.com/ecamp/ecamp3/pull/5170/commits/fc71576bb37bd0143a1db3a4b7eb7dc4b4138032

File: frontend/src/components/activity/content/storyboard/StoryboardRowDefault.vue [hunk 1]
```diff
@@ -7,8 +7,8 @@
         class="drag-and-drop-handle"
         :disabled="isLastSection"
         :aria-label="$tc('global.button.move')"
-        @keydown.down="$emit('moveDown', itemKey)"
-        @keydown.up="$emit('moveUp', itemKey)"
+        @keydown.down="$emit('move-down', itemKey)"
```
Comment (BacLuc) [inline frontend/src/components/activity/content/storyboard/StoryboardRowDefault.vue:10]: @manuelmeister 
Should it be possible to move the storyboard entries here around with the keyboard?
how do i test this?

**Review by pmattmann — APPROVED** [review]

File: frontend/src/App.vue [hunk 2]
```diff
@@ -42,7 +42,7 @@ export default {
       }
     }
   },
-  destroyed() {
+  unmounted() {
```
Comment (usu) [inline frontend/src/App.vue]: According to vue2 docs, there are no lifecycle hooks `unmounted` or `beforeUnmount`. Did you check if this code is still called?

https://v2.vuejs.org/v2/guide/instance#Lifecycle-Diagram
https://v2.vuejs.org/v2/api/#Options-Lifecycle-Hooks

File: frontend/src/components/form/base/EDatePicker.vue [hunk 3]
```diff
@@ -20,6 +20,7 @@ Displays a field as a date picker (can be used with v-model)
   >
     <template #default="picker">
       <v-date-picker
+        :picker-date="pickerMonth"
```
Comment (usu) [inline frontend/src/components/form/base/EDatePicker.vue]: Why is the .sync modifier not needed anymore?
(same question for the other places, where the .sync modifier was removed)

**Review by usu — CHANGES_REQUESTED** [review]

File: frontend/src/components/form/base/EDatePicker.vue (hunk 3 above)
Comment (usu) [reply frontend/src/components/form/base/EDatePicker.vue]: In vue3 there's a replacement for multiple 2-way-binding properties (argument for v-model):
https://v3-migration.vuejs.org/breaking-changes/v-model

But I guess this is not working in vue2.

File: frontend/src/App.vue (hunk 2 above)
Comment (BacLuc) [reply frontend/src/App.vue]: i reverted that change.
They were not called

File: frontend/src/components/form/base/EDatePicker.vue (hunk 3 above)
Comment (BacLuc) [reply frontend/src/components/form/base/EDatePicker.vue]: I remember vaguely that i had some problems with the .sync modifier.
But i could not reproduce it, so i added the .sync modifier, but changed the attribute order.

File: frontend/src/components/program/picasso/PicassoEntry.vue [hunk 4]
```diff
@@ -265,7 +265,7 @@ export default {
     this.scrollHeight = this.$el.scrollHeight
     window.addEventListener('resize', this.onResize)
   },
-  destroyed() {
+  unmounted() {
```
Comment (usu) [inline frontend/src/components/program/picasso/PicassoEntry.vue]: This one got forgotten to revert.

```suggestion
  destroyed() {
```

**Review by usu — COMMENTED** [review]
Other than the comment below, looks good to me 👍

**Review by usu — APPROVED** [review]

**Comment by BacLuc** [comment]
More rule applications were here: #5122

**Pull request merged**